vertical-pod-autoscaler/e2e/v1/admission_controller.go

```diff
@@ -517,6 +518,9 @@ var _ = AdmissionControllerE2eDescribe("Admission-controller", func() {
 })
 
 func startDeploymentPods(f *framework.Framework, deployment *appsv1.Deployment) *apiv1.PodList {
+	// Etcd watch can lag depending on etcd object count and etcd resource usage.
+	// We assume that ectd can lag up to 5 seconds.
+	const etcdWatchLag = 5 * time.Second
```

Comment:
Why 5 second delay? Will it be enough?

Reply:
Should be enough. Imho the lag should be  max 1s, so 5s is a little bit over the top.
In this case we want to focus on specific scenarios, so it's not a problem to wait little longer.